curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-hash-table.english.md

```diff
@@ -32,9 +32,9 @@ tests:
   - text: The HashTable should have a lookup method.
     testString: assert((function() { var test = false; if (typeof HashTable !== 'undefined') { test = new HashTable() }; return ((typeof test.lookup) === 'function')})());
   - text: The add method should add key value pairs and the lookup method should return the values associated with a given key.
-    testString: assert((function() { var test = false; if (typeof HashTable !== 'undefined') { test = new HashTable() }; test.add('key', 'value'); return (test.lookup('key') === 'value')})());
+    testString: assert((function(){ var test = false; if (typeof HashTable !== 'undefined') { test = new HashTable(); }; test.add('key', 'value'); test.lookup = function(key){ var theHash = hash(key); if (this.collection.hasOwnProperty(theHash)) { return this.collection[theHash][key]; } return null }; var lookup = test.lookup('key'); test.lookup = null; return (lookup === 'value')})());
   - text: The remove method should accept a key as input and should remove the associated key value pair.
-    testString: assert((function() { var test = false; if (typeof HashTable !== 'undefined') { test = new HashTable() }; test.add('key', 'value'); test.remove('key'); return (test.lookup('key') === null)})());
+    testString: assert((function(){ var test = false; if (typeof HashTable !== 'undefined') { test = new HashTable() }; test.add('key', 'value'); test.remove('key'); test.lookup = function(key){ var theHash = hash(key); if (this.collection.hasOwnProperty(theHash)) { return this.collection[theHash][key]; } return null }; var lookup = test.lookup('key'); test.lookup = null; return (lookup === null)})());
```

Comment:
This rejects code that doesn't delete `theHash` from `this.collection`, but that's I think that's probably fine.

Reply:
I wasn't sure what you meant at first @ojeytonwilliams - but you're saying that if the users code doesn't delete the hashed key, it will get rejected.

So when you add something, you get something like...
```
307: {
  key: 'value'
}
```

and the 307 might remain after the `remove`.

I feel like that is something that should be addressed, since it doesn't say anything about removing the hashed code/key from the collection (If I'm understanding all this right). And I think that it would be better to adjust the test to allow for that, rather than trying to rewrite things.

I suppose it might depend on how users go about adding to the collection. But [the code I tested from the forum](https://www.freecodecamp.org/forum/t/having-trouble-with-create-a-hash-table-challenge/324284) doesn't pass.

I think just adding this to the test would pass...
```
if (this.collection.hasOwnProperty(theHash)[key]) { 
  return this.collection[theHash][key];
} 
```
unless users go about storing the values directly in the hashed key.

Reply:
Sorry, I wasn't clear.  If you have a data structure

```js
this.collection = {
  307: {key: value}
}
```
and `remove` deletes just the key value pair then you'll have

```js
this.collection = {
  307: {}
}
```
and `lookup(key-that-hashes-to-307)` is `undefined` not `null`, so this test rejects it.  A simple change to the test that would allow this is to use `return !lookup`.

> I feel like that is something that should be addressed, since it doesn't say anything about removing the hashed code/key from the collection 

That's fair.  The tests should be explicit about what they want.

Reply:
Yea, I think that should be added. The forum code I linked to is the code that the [original issue](https://github.com/freeCodeCamp/freeCodeCamp/issues/37577) was trying to accommodate. And it doesn't pass without something else here.

Reply:
So I should add 
```
if (this.collection.hasOwnProperty(theHash)[key]) { 
  return this.collection[theHash][key];
}
```
in the lookup for the remove test?